src/main/scala/org/stormenroute/mecha/package.scala

```diff
@@ -432,4 +432,113 @@ package object mecha {
     def default(v: =>T): Input.Query[T] = Input.default(query)(v)
   }
 
+  private[mecha] object GitIgnore {
+    import java.nio.charset.StandardCharsets._
+    import java.nio.file.{FileSystems, Paths, Files, Path}
+    import java.nio.file.StandardOpenOption._
+    import scala.collection.JavaConversions._
+
+    val log = MechaLog.Println
+
+    def readFile(file: Path): Seq[String] = {
+      Files.readAllLines(file, UTF_8)
+    }
+
+    def writeFile(file: Path, content: Seq[String]) = {
+      Files.write(file, content, UTF_8, WRITE, CREATE)
+    }
+
+    def ignore(toIgnore: String, gitIgnoreFilePath: Path, gitExcludeFilePath: Path) = {
+      val gitIgnore = if (Files.exists(gitIgnoreFilePath)) readFile(gitIgnoreFilePath) else Seq()
+      val gitExclude = if (Files.exists(gitExcludeFilePath)) readFile(gitExcludeFilePath) else Seq()
+
+      addIgnore(toIgnore, gitExclude, gitIgnore ++ gitExclude).fold(
+          {
+            case Whitelists(patterns) =>
+              log.warn(s"The new repo can not be ignored by Git. It is whitelisted by the following patterns: $patterns")
+            case Blacklists(patterns) =>
+              log.info(s"The new repo is already ignored via the following pattern(s): $patterns")
+          },
+          b => writeFile(gitExcludeFilePath, b))
+    }
+
+    sealed trait Patterns
+    case class Whitelists(patterns: Seq[WhitelistPattern]) extends Patterns
+    case class Blacklists(patterns: Seq[BlacklistPattern]) extends Patterns
+
+    def addIgnore(toAdd: String, addTo: Seq[String], check: Seq[String]): Either[Patterns, Seq[String]] = {
```

Comment:
Perhaps a less ambiguous naming could be: `(lineToAdd: String, excludeLines: Seq[String], allLines: Seq[String])`